plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/ServerConfig.java

```diff
@@ -0,0 +1,232 @@
+        keepAlive = FLIGHT_THREAD_POOL_KEEP_ALIVE.get(settings);
+    }
+
+    /**
+     * Checks if SSL/TLS is enabled for the Flight server.
+     *
+     * @return true if SSL is enabled, false otherwise
+     */
+    public static boolean isSslEnabled() {
+        return enableSsl;
+    }
+
+    /**
+     * Gets the thread pool executor builder configured for the Flight server.
+     *
+     * @return The configured ScalingExecutorBuilder instance
+     */
+    public static ScalingExecutorBuilder getServerExecutorBuilder() {
+        return new ScalingExecutorBuilder(FLIGHT_SERVER_THREAD_POOL_NAME, threadPoolMin, threadPoolMax, keepAlive);
+    }
+
+    /**
+     * Gets the thread pool executor builder configured for the Flight server.
+     *
+     * @return The configured ScalingExecutorBuilder instance
+     */
+    public static ScalingExecutorBuilder getClientExecutorBuilder() {
+        return new ScalingExecutorBuilder(FLIGHT_CLIENT_THREAD_POOL_NAME, threadPoolMin, threadPoolMax, keepAlive);
+    }
+
+    /**
+     * Returns a list of all settings managed by this configuration class.
+     *
+     * @return List of Setting instances
+     */
+    public static List<Setting<?>> getSettings() {
+        return new ArrayList<>(
+            Arrays.asList(
+                ARROW_ALLOCATION_MANAGER_TYPE,
+                ARROW_ENABLE_NULL_CHECK_FOR_GET,
+                ARROW_ENABLE_DEBUG_ALLOCATOR,
+                ARROW_ENABLE_UNSAFE_MEMORY_ACCESS,
+                ARROW_SSL_ENABLE
+            )
+        ) {
+            {
+                addAll(Netty4Configs.getSettings());
+            }
+        };
+    }
+
+    static EventLoopGroup createELG(String name, int eventLoopThreads) {
+
+        return Epoll.isAvailable()
+            ? new EpollEventLoopGroup(eventLoopThreads, new DefaultThreadFactory(name, true))
+            : new NioEventLoopGroup(eventLoopThreads, new DefaultThreadFactory(name, true));
+    }
+
+    static Class<? extends Channel> serverChannelType() {
+        return Epoll.isAvailable() ? EpollServerSocketChannel.class : NioServerSocketChannel.class;
+    }
+
+    static Class<? extends Channel> clientChannelType() {
+        return Epoll.isAvailable() ? EpollSocketChannel.class : NioSocketChannel.class;
+    }
+
+    private static class Netty4Configs {
+        public static final Setting<Integer> NETTY_ALLOCATOR_NUM_DIRECT_ARENAS = Setting.intSetting(
+            "io.netty.allocator.numDirectArenas",
+            1, // TODO - 2 * the number of available processors; to be confirmed and set after running benchmarks
+            1,
+            Setting.Property.NodeScope
+        );
+
+        @SuppressForbidden(reason = "required for netty allocator configuration")
+        public static void init(Settings settings) {
+            System.setProperty("io.netty.allocator.numDirectArenas", Integer.toString(NETTY_ALLOCATOR_NUM_DIRECT_ARENAS.get(settings)));
```

Comment:
:( still the problem, may impact `NettyAllocator` (check please netty4 transport)

Reply:
removed it now and made it as part of mandatory check when feature flag is set.